test/Microsoft.ML.Functional.Tests/Debugging.cs

```diff
@@ -0,0 +1,214 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using Microsoft.ML.Data;
+using Microsoft.ML.Functional.Tests.Datasets;
+using Microsoft.ML.RunTests;
+using Microsoft.ML.TestFramework;
+using Microsoft.ML.Trainers;
+using Microsoft.ML.Transforms.Text;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.ML.Functional.Tests
+{
+    public class Debugging : BaseTestClass
+    {
+        public Debugging(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        /// <summary>
+        /// Debugging: The individual pipeline steps can be inspected to see what is happening to 
+        /// data as it flows through.
+        /// </summary>
+        /// <remarks>
+        /// It should, possibly through the debugger, be not such a pain to actually
+        /// see what is happening to your data when you apply this or that transform. For example, if I
+        /// were to have the text "Help I'm a bug!" I should be able to see the steps where it is
+        /// normalized to "help i'm a bug" then tokenized into ["help", "i'm", "a", "bug"] then
+        /// mapped into term numbers [203, 25, 3, 511] then projected into the sparse
+        /// float vector {3:1, 25:1, 203:1, 511:1}, etc. etc.
+        /// </remarks>
+        [Fact]
+        void InspectIntermediatePipelineSteps()
+        {
+            var mlContext = new MLContext(seed: 1);
+
+            var data = mlContext.Data.LoadFromEnumerable<TweetSentiment>(
+                new TweetSentiment[]
+                {
+                    new TweetSentiment { Sentiment = true, SentimentText = "I love ML.NET." },
+                    new TweetSentiment { Sentiment = true, SentimentText = "I love TLC." },
```

Comment:
>TLC [](start = 83, length = 3)

?

Reply:
Tiny little cakes?

---
In reply to: [265409347](https://github.com/dotnet/machinelearning/pull/2937#discussion_r265409347) [](ancestors = 265409347)

Reply:
lol, Tender Loving Care

---
In reply to: [265411100](https://github.com/dotnet/machinelearning/pull/2937#discussion_r265411100) [](ancestors = 265411100,265409347)